Sage/carbon-codemod
# fix: address broken internal git hooks

### Proposed behaviour

- Update `husky` to version 8.x
- Address `husky` breaking changes between versions 4.x to 8.x
- Update all dependencies, including these to their latest versions:
    - `@commitlint/cli` from version 13.1.0 to 17.6.1
    - `@commitlint/config-conventional` from version 13.2.0 to 17.6.1
    - `@semantic-release/changelog` from version 5.0.1 to 6.0.3
    - `@semantic-release/git` from version 9.0.0 to 10.0.1

### Current behaviour

- Many packages are outdated and require upgrading.
- Commit checks are not running after `git commit` - since we accidentally upgraded `husky` without addressing the introduced breaking changes (see PR #55). This means its possible to write commit messages which `semantic-release` would not recognise.

### Checklist




### Testing instructions

To check post-commit actions are being triggered correctly, when making a commit, check the following actions are completed after `git commit`:
   - Prettier is run when javascript file is changed
   - Tests are run
   - Commitlint is run on commit message
